packages/nft-checkout/src/operations/evm/helpers/payment.ts

```diff
@@ -2,86 +2,158 @@ import { BN } from '@distributedlab/tools'
 import type { Token } from '@rarimo/bridge'
 import { tokenFromChain } from '@rarimo/bridge'
 import type { IProvider } from '@rarimo/provider'
-import type { BridgeChain } from '@rarimo/shared'
-import { Amount } from '@rarimo/shared'
 import type {
-  BalanceResult,
-  Token as TokenInfo,
-} from 'ethereum-erc20-token-balances-multicall'
-import { getBalancesForEthereumAddress } from 'ethereum-erc20-token-balances-multicall'
+  BridgeChain,
+  InternalAccountBalance,
+  TokenSymbol,
+} from '@rarimo/shared'
+import {
+  Amount,
+  ChainNames,
+  loadAccountBalances,
+  parseTokenId,
+  toLowerCase,
+} from '@rarimo/shared'
+import type { Swapper } from '@rarimo/swap'
 
 import { paymentTokenFromToken } from '@/entities'
-import type { PaymentToken } from '@/types'
+import type { CheckoutOperationParams, PaymentToken } from '@/types'
 
-const mapTokenBalances = (
-  supportedTokens: Token[],
-  balances: BalanceResult,
-): PaymentToken[] => {
-  if (!balances.tokens) return []
+import { estimate, getNativeAmountIn, isSameChainOperation } from './estimator'
+import { NATIVE_TOKEN_ADDRESS } from './tokens'
+
+const INTERNAL_TOKENS_MAP: { [key in ChainNames]?: string } = {
+  [ChainNames.Goerli]: '2',
+  [ChainNames.Sepolia]: '3',
+  [ChainNames.Fuji]: '4',
+  [ChainNames.Chapel]: '5',
+}
+
+export const getPaymentTokensWithPairs = async (
+  provider: IProvider,
+  params: CheckoutOperationParams,
+  tokens: Token[],
+  targetToken: Token,
+  chainFrom: BridgeChain,
+) => {
+  const paymentTokens = await getPaymentTokens(provider, chainFrom, tokens)
+
+  const targetTokenSymbol = toLowerCase(targetToken.symbol)
+
+  const paymentTokensWithoutTarget = paymentTokens.filter(
+    i => toLowerCase(i.symbol) !== targetTokenSymbol,
+  )
+
+  const estimatedPrices = await Promise.allSettled(
+    paymentTokensWithoutTarget.map(i =>
+      estimate(provider, tokens, i, params, targetToken),
+    ),
+  )
+
+  return estimatedPrices.reduce<PaymentToken[]>((acc, i) => {
```

Comment:
let's use `promise` instead of `i` is better for readability and consistency with reduce below

I've spent some time to understand why estimated price has status and where it came from